charts/internal/shoot-storageclasses/templates/storageclasses.yaml

```diff
@@ -0,0 +1,63 @@
+{{- if not .Values.useLegacyProvisioner }}
+---
+apiVersion: {{ include "storageclassversion" . }}
+kind: StorageClass
+metadata:
+  name: default
+  annotations:
+    storageclass.kubernetes.io/is-default-class: "true"
+provisioner: disk.csi.azure.com
+parameters:
+  skuName: Standard_LRS
+  kind: managed
+volumeBindingMode: WaitForFirstConsumer
+
+---
+apiVersion: {{ include "storageclassversion" . }}
+kind: StorageClass
+metadata:
+  name: managed-standard-hdd
+provisioner: disk.csi.azure.com
+parameters:
+  skuName: Standard_LRS
+  kind: managed
+volumeBindingMode: WaitForFirstConsumer
+allowVolumeExpansion: true
+
+{{- if semverCompare ">= 1.13" .Capabilities.KubeVersion.GitVersion }}
+---
+apiVersion: {{ include "storageclassversion" . }}
+kind: StorageClass
+metadata:
+  name: managed-standard-ssd
+provisioner: disk.csi.azure.com
+parameters:
+  skuName: StandardSSD_LRS
+  kind: managed
+volumeBindingMode: WaitForFirstConsumer
+allowVolumeExpansion: true
+{{- end }}
+
+---
+apiVersion: {{ include "storageclassversion" . }}
+kind: StorageClass
+metadata:
+  name: managed-premium-ssd
+provisioner: disk.csi.azure.com
+parameters:
+  skuName: Premium_LRS
+  kind: managed
+volumeBindingMode: WaitForFirstConsumer
+allowVolumeExpansion: true
+
+---
+apiVersion: {{ include "storageclassversion" . }}
+kind: StorageClass
+metadata:
+  name: files
+provisioner: file.csi.azure.com
```

Comment:
The azurefile-csi-driver has issue during mount - ref https://github.com/kubernetes-sigs/azurefile-csi-driver/issues/247.

Reply:
@ialidzhikov how do we continue here?

Reply:
From the issue above, I think the feedback was that azurefile-csi-driver running in the Shoot  needs identity/credentials to access the storage account.

Reply:
OK, thanks. I'll check with the maintainers.

Reply:
(Already discussed via phone, just writing it down here so that others are aware as well): With the latest changes of this PR we just need a new version of the Azure-File CSI driver and then this PR should be good to go.

Reply:
So we tested with @rfranzke the master branch of azurefile-csi-driver and it was working fine. So we just need new release of azurefile-csi-driver and we are good to go.

Reply:
FYI: Release planned for this week.